FOGL-2751: use compiled sqlite static library in /tmp/foglamp-sqlite3…

pintomax opened the pull request:
…-pkg

FOGL-2751: use compiled sqlite static library in /tmp/foglamp-sqlite3-pkg

cd /tmp/
git clone https://github.com/foglamp/foglamp-sqlite3-pkg.git
cd foglamp-sqlite3-pkg
cd src
./configure --enable-shared=false --enable-static=true --enable-static-shell CFLAGS="-DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_LOAD_EXTENSION -DSQLITE_ENABLE_COLUMN_METADATA -fno-common -fPIC"
autoreconf -f -i
make

cd to foglamp dir
make

pintomax committed:
FOGL-2751: use compiled sqlite static library in /tmp/foglamp-sqlite3-pkg

FOGL-2751: use compiled sqlite static library in /tmp/foglamp-sqlite3-pkg

cd /tmp/
git clone https://github.com/foglamp/foglamp-sqlite3-pkg.git
cd foglamp-sqlite3-pkg/
./configure --enable-shared=false --enable-static=true --enable-static-shell CFLAGS="-DSQLITE_ENABLE_JSON1 -DSQLITE_ENABLE_LOAD_EXTENSION -DSQLITE_ENABLE_COLUMN_METADATA -fno-common -fPIC"
autoreconf -f -i
make

cd to foglamp dir
make

MarkRiddoch reviewed (commented):
Looks fine to me. Have you tested this code on Debian and/or Raspbian to make sure there are no regressions?

pintomax committed:
Makefile fix

Makefile fix

pintomax commented:
I see an issue with make -j2

pintomax committed:
Fix for local copy of sqlite cmd tool and make -j2

Fix for local copy of sqlite cmd tool and make -j2

pintomax committed:
Fix plugin_dir

Fix plugin_dir

pintomax committed:
First implementation of os detection in requirements.sh

First implementation of os detection in requirements.sh

In case of CentOS|RHEL we compile sqlite3 static library in /tmp/foglamp-sqlite3-pkg

pintomax committed:
Fix for RHEL systems

Fix for RHEL systems

pintomax committed:
Additional fixes for RHEL

Additional fixes for RHEL

pintomax commented:
@Singhal-Vaibhav I've tested:

[ec2-user@ip-10-0-0-93 FogLAMP]$  sh requirements.sh

and

ubuntu@ubuntu16-server-virt: FogLAMP$ sh requirements.sh

AmandeepArora commented on requirements.sh:
```diff
@@ -17,19 +17,62 @@
+is_rhel=`(lsb_release -ds 2>/dev/null || cat /etc/*release 2>/dev/null | head -n1 || uname -om) | egrep '(Red Hat|CentOS)' || echo ""`
+if [ "${is_rhel}" != "" ]; then
+	echo "Platform is ${is_rhel}"
+	sudo yum check-update
```
This script runs only with sudo, so the "sudo" prefix for these yum commands are not required. Same for apt commands below also.

AmandeepArora reviewed (commented):
Looks good. A minor comment.

pintomax committed:
Merge branch 'develop' into FOGL-2751-sqlite3-fix

Singhal-Vaibhav commented on requirements.sh:
```diff
@@ -17,19 +17,62 @@
+	echo "Platform is ${is_rhel}"
+	sudo yum check-update
+	sudo yum update -y
+	sudo yum-config-manager --enable 'Red Hat Enterprise Linux Server 7 RHSCL (RPMs)'
```
is_rhel is a generic check for all RHEL and CentOS, however this is not available on rhel 8 and centos and the script will fail in case this runs on these environments
``` Error: No matching repo to modify: Red Hat Enterprise Linux Server 7 RHSCL (RPMs).```

Singhal-Vaibhav commented:
Also, foglamp start does not work. I was expecting them to work when we have installed rh-python36 in requirements.sh, steps:
1. Provision a fresh rhel 7 machine
2. sudo yum check-update
3. sudo yum update -y
4. sudo yum install git
5. git clone https://github.com/foglamp/FogLAMP.git
6. cd FogLAMP
7. git checkout -b FOGL-2751-sqlite3-fix origin/FOGL-2751-sqlite3-fix
8. sudo sh requirements.sh 
9. sudo make install
10. export FOGLAMP_ROOT=/usr/local/foglamp/
11. $FOGLAMP_ROOT/bin/foglamp start
Python interpreter not found, FogLAMP cannot start.

Singhal-Vaibhav commented:
On ubuntu16 and 18, foglamp start does not work, steps as follows:
1. Provision a fresh ubuntu 16/ 18 machine
2. sudo apt update
3. sudo apt upgrade
4. sudo apt install git
5. git clone https://github.com/foglamp/FogLAMP.git
6. cd FogLAMP
7. git checkout -b FOGL-2751-sqlite3-fix origin/FOGL-2751-sqlite3-fix
8. sudo sh requirements.sh
9. sudo make install
10. export FOGLAMP_ROOT=/usr/local/foglamp/
11. $FOGLAMP_ROOT/bin/foglamp start
Syslogs:
```
May  9 17:23:05 foglamp-VirtualBox FogLAMP Storage[27090] INFO: script.plugin.storage.sqlite: SQLite 3 database '/usr/local/foglamp/data/foglamp.db' not found.
```
Following the same steps, foglamp starts on develop branch.

pintomax commented:
@Singhal-Vaibhav thanks for spotting that. Working on a fix.

pintomax committed:
Fix for sqite3 cmd tool location

Fix for sqite3 cmd tool location

pintomax commented:
@Singhal-Vaibhav added check fro sqlite3 command line.
It was a regression bug.

Just tested in RHEL7 and Ubuntu:
(a) git clone; make export FOGLAMP_ROOT, scripts/foglamp start
(b) git clone; sudo make install; /usr/local/foglamp/bin/foglamp

pintomax commented:
Tested on Raspberry Pi3:
make, start
sudo make install, /usr/local/foglamp/bin/foglamp start

MarkRiddoch commented:
Given the nature of this let's get it merged as soon as we can so there is a chance that any issues can be picked up before the weekend. I do not like the idea of having a major checkin last thing on a Friday afternoon if we can avoid it.

sstefano69 reviewed (approved):
tested on my local Centos Vbox machine and it worked fine.
